tests/cross_functional/system_test/test_mcg_replication_with_disruptions.py

```diff
@@ -404,3 +419,333 @@ def test_log_based_replication_with_disruptions(
+        reduce_replication_delay()
+
+        # Setup two buckets with bi-directional replication enabled
+        # deletion sync disabled
+        bucketclass_dict = {
+            "interface": "OC",
+            "backingstore_dict": {"aws": [(1, "eu-central-1")]},
+        }
+        mockup_logger_source, mockup_logger_target, bucket_1, bucket_2 = (
+            aws_log_based_replication_setup(
+                bucketclass_dict=bucketclass_dict,
+                bidirectional=True,
+                prefix_source=prefix_1,
+                prefix_target=prefix_2,
+                deletion_sync=False,
+            )
+        )
+
+        # Upload object and verify that bucket replication works
+        logger.info(f"Uploading object {object_key} to the bucket {bucket_1.name}")
+        self.upload_objects_with_retry(
+            mcg_obj_session,
+            bucket_1,
+            bucket_2,
+            mockup_logger_source,
+            test_directory_setup.origin_dir,
+            pattern=object_key,
+            prefix=prefix_1,
+        )
+
+        # Enable object versioning on both the buckets
+        s3_put_bucket_versioning(mcg_obj_session, bucket_1.name)
+        s3_put_bucket_versioning(mcg_obj_session, bucket_2.name)
+        logger.info("Enabled object versioning for both the buckets")
+
+        # Enable sync versions in both buckets replication policy
+        replication_1 = json.loads(get_replication_policy(bucket_name=bucket_2.name))
+        replication_2 = json.loads(get_replication_policy(bucket_name=bucket_1.name))
+        replication_1["rules"][0]["sync_versions"] = True
+        replication_2["rules"][0]["sync_versions"] = True
+
+        update_replication_policy(bucket_2.name, replication_1)
+        update_replication_policy(bucket_1.name, replication_2)
+        logger.info(
+            "Enabled sync versions in the replication policy for both the buckets"
+        )
+
+        # Update previously uploaded object with new data and new version
+        self.upload_objects_with_retry(
+            mcg_obj_session,
+            bucket_2,
+            bucket_1,
+            mockup_logger_target,
+            test_directory_setup.origin_dir,
+            pattern=object_key,
+            prefix=prefix_2,
+        )
+        logger.info(
+            f"Updated object {object_key} with new version data in bucket {bucket_2.name}"
+        )
+
+        wait_for_object_versions_match(
+            mcg_obj_session,
+            awscli_pod_session,
+            bucket_1.name,
+            bucket_2.name,
+            obj_key=f"{prefix_2}/{object_key}",
+        )
+        logger.info(
+            f"Replication works from {bucket_2.name} to {bucket_1.name} and has all the versions of object {object_key}"
+        )
+
+        # Will perform disruptive operations and object uploads, version verifications
+        # parallely.
+        with ThreadPoolExecutor(max_workers=1) as executor:
```

Comment:
you are setting the max_workers to 1, does it really provide the expected concurrency?

Reply:
yes. We just need one more thread apart form the main thread